test/e2e/getenvoy_extension_build_test.go

```diff
@@ -15,70 +15,48 @@
 package e2e_test
 
 import (
-	"os"
 	"path/filepath"
+	"testing"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/ginkgo/extensions/table"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 
-	workspaces "github.com/tetratelabs/getenvoy/pkg/extension/workspace"
-	toolchains "github.com/tetratelabs/getenvoy/pkg/extension/workspace/toolchain"
 	e2e "github.com/tetratelabs/getenvoy/test/e2e/util"
 )
 
-var _ = Describe("getenvoy extension build", func() {
-
-	type testCase e2e.CategoryLanguageTuple
-
-	testCases := func() []TableEntry {
-		testCases := make([]TableEntry, 0)
-		for _, combination := range e2e.GetCategoryLanguageCombinations() {
-			testCases = append(testCases, Entry(combination.String(), testCase(combination)))
-		}
-		return testCases
+// TestGetEnvoyExtensionBuild runs the equivalent of "getenvoy extension build" for a matrix of extension.Categories and
+// extension.Languages. "getenvoy extension init" is a prerequisite, so run first.
+//
+// Note: "getenvoy extension build" can be extremely slow due to implicit responsibilities such as downloading modules
+// or compilation. This uses Docker, so changes to the Dockerfile or contents like "commands.sh" effect performance.
+//
+// Note: Pay close attention to values of util.E2E_BUILTIN_TOOLCHAIN_CONTAINER_OPTIONS as these can change assumptions.
+// CI may override this to set HOME or CARGO_HOME (rust) used by "getenvoy" and effect its execution.
+func TestGetEnvoyExtensionBuild(t *testing.T) {
+	const extensionName = "getenvoy_extension_build"
+	requireEnvoyBinaryPath(t) // Ex. After running "make bin", E2E_GETENVOY_BINARY=$PWD/build/bin/darwin/amd64/getenvoy
```

Comment:
maybe we'd better move `requireEnvoyBinaryPath` to `TestMain` since it's enough to check at the startup time?

Reply:
so that we don't need to copy and paste this in every e2e :smile:

Reply:
sgtm